packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +75,14 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    // TODO(eseidel): This clearly isn't correct.  Not sure what would be.
-    // Should we use the ios images on ios?
-    String density = 'drawable-xxhdpi';
+    String density = new BreakpointQualifier(
+      candidates: _kIconDensities,
+      value: MediaQuery.of(context).devicePixelRatio
+    ).resolve();
```

Comment:
It's strange to me that we'd need to create a new BreakpointQualifier every time we go through build.  The icon densities are constant and the devicePixelRatio is almost always the same.  I would have expected there to be a retained static object that held the _kIconDensities static and that you could query with the devicePixelRatio.  It would then have a cache so that when you're repeatedly querying for the same devicePixelRatio (which is the common case), you get back the same answer quickly.